From 4317bf85f429a021addc59c4356a24c4b3b6d409 Mon Sep 17 00:00:00 2001
From: 0xvd <0xvd12@gmail.com>
Date: Tue, 24 Feb 2026 23:56:05 +0530
Subject: [PATCH 1/7] [ie/blogger] Fix extractor

---
 yt_dlp/extractor/blogger.py | 111 ++++++++++++++++++++++++++++++------
 1 file changed, 92 insertions(+), 19 deletions(-)

diff --git a/yt_dlp/extractor/blogger.py b/yt_dlp/extractor/blogger.py
index 6ff72e70d2ca..6fc1bd9ee3ba 100644
--- a/yt_dlp/extractor/blogger.py
+++ b/yt_dlp/extractor/blogger.py
@@ -1,11 +1,14 @@
+import random
+
 from .common import InfoExtractor
 from ..utils import (
-    mimetype2ext,
     parse_duration,
     parse_qs,
     str_or_none,
-    traverse_obj,
+    url_or_none,
+    urlencode_postdata,
 )
+from ..utils.traversal import traverse_obj
 
 
 class BloggerIE(InfoExtractor):
@@ -19,7 +22,17 @@ class BloggerIE(InfoExtractor):
             'id': 'BLOGGER-video-3c740e3a49197e16-796',
             'title': 'BLOGGER-video-3c740e3a49197e16-796',
             'ext': 'mp4',
-            'duration': 76.068,
+            'duration': 76.006,
+            'thumbnail': r're:https?://i9\.ytimg\.com/vi_blogger/.+',
+        },
+    }, {
+        'url': 'https://www.blogger.com/video.g?token=AD6v5dxQvB2soRxjzQ6NfKtsKV42CPEC905mnK-zop3trOP32hF4Jb3895aWvR85zvR8Ks5U5Zyyvrkj6nQuOg_9o0HK4C1Iclp-eRxHveFRdhACzMi67UeFS9CO-2Z9r_FCKxfOPA',
+        'md5': '309f9a5604ebd272dd2e1538998a0f40',
+        'info_dict': {
+            'id': 'BLOGGER-video-6f028977527fdc05-338',
+            'ext': 'mp4',
+            'title': 'BLOGGER-video-6f028977527fdc05-338',
+            'duration': 75.966,
             'thumbnail': r're:https?://i9\.ytimg\.com/vi_blogger/.+',
         },
     }]
@@ -27,30 +40,90 @@ class BloggerIE(InfoExtractor):
         'url': 'https://blog.tomeuvizoso.net/2019/01/a-panfrost-milestone.html',
         'md5': 'f1bc19b6ea1b0fd1d81e84ca9ec467ac',
         'info_dict': {
-            'id': 'BLOGGER-video-3c740e3a49197e16-12203',
+            'id': 'BLOGGER-video-3c740e3a49197e16-12913',
             'ext': 'mp4',
-            'title': 'BLOGGER-video-3c740e3a49197e16-12203',
-            'duration': 76.068,
+            'title': 'BLOGGER-video-3c740e3a49197e16-12913',
+            'duration': 76.006,
             'thumbnail': r're:https?://i9\.ytimg\.com/vi_blogger/.+',
         },
     }]
 
+    # Source https://www.blogger.com/_/scs/mss-static/_/js/k=boq-blogger.BloggerVideoPlayerUi.en_US.6PupVFhoDlA.2018.O/ck=boq-blogger.BloggerVideoPlayerUi.0lo6xI35Cfs.L.B1.O/am=AAAAwA4/d=1/exm=_b,_tp/excm=_b,_tp,videoplayerview/ed=1/wt=2/ujg=1/rs=AEy-KP3OJHU74ptfWOTb5LeMqh3Kxak-yA/ee=EVNhjf:pw70Gc;EmZ2Bf:zr1jrb;JsbNhc:Xd8iUd;K5nYTd:ZDZcre;LBgRLc:XVMNvd;Me32dd:MEeYgc;NJ1rfe:qTnoBf;NPKaK:PVlQOd;Pjplud:EEDORb;QGR0gd:Mlhmy;SNUn3:ZwDk9d;ScI3Yc:e7Hzgb;Uvc8o:VDovNc;YIZmRd:A1yn5d;a56pNe:JEfCwb;cEt90b:ws9Tlc;dIoSBb:SpsfSb;dowIGb:ebZ3mb;eBAeSb:zbML3c;iFQyKf:QIhFr;lOO0Vd:OTA3Ae;oGtAuc:sOXFj;qQEoOc:KUM7Z;qafBPd:yDVVkb;qddgKe:xQtZb;wR5FRb:siKnQd;yxTchf:KUM7Z/dti=1/m=OXnWq
+    # current varianle name (s1) and format table name (r1).
+    IF_TABLE = {
+        7: {'width': 320, 'height': 240},
+        18: {'width': 640, 'height': 360},
+        22: {'width': 1280, 'height': 720},
+        37: {'width': 1920, 'height': 1080},
+        13: {'width': 256, 'height': 144},  # Source https://gist.github.com/MartinEesmaa/2f4b261cb90a47e9c41ba115a011a4aa#legacy-non-dash
+    }
+
     def _real_extract(self, url):
         token_id = self._match_id(url)
-        webpage = self._download_webpage(url, token_id)
-        data_json = self._search_regex(r'var\s+VIDEO_CONFIG\s*=\s*(\{.*)', webpage, 'JSON data')
-        data = self._parse_json(data_json.encode().decode('unicode_escape'), token_id)
-        streams = data['streams']
-        formats = [{
-            'ext': mimetype2ext(traverse_obj(parse_qs(stream['play_url']), ('mime', 0))),
-            'url': stream['play_url'],
-            'format_id': str_or_none(stream.get('format_id')),
-        } for stream in streams]
+
+        fsid = random.randint(-9999999999999999999, 9999999999999999999)  # Blogger require 19 digit number can be negative or positive
+
+        data, _ = self._download_webpage_handle(
+            'https://www.blogger.com/_/BloggerVideoPlayerUi/data/batchexecute',
+            token_id,
+            note='Downloading blogger batch information',
+            headers={
+                'Referer': 'https://www.blogger.com/',
+            },
+            query={
+                'rpcids': 'WcwnYd',
+                'f.sid': fsid,
+                'bl': 'boq_bloggeruiserver_20260218.01_p0',
+                'hl': 'en-US',
+                'rt': 'c',
+            },
+            data=urlencode_postdata({
+                'f.req': f'[[["WcwnYd","[\\"{token_id}\\",\\"\\",0]",null,"generic"]]]',
+            }),
+        )
+
+        wrb_data = self._search_regex(r'WcwnYd\s*",\s*"(\[\[.*?\]\])', data, 'wrb.fr')
+        data = self._parse_json(self._parse_json(wrb_data, None)[0][2], None)
+
+        formats = []
+        for fmtl in traverse_obj(data, (2), default=[]):
+            if not isinstance(fmtl, list):
+                continue
+            fmt_url = traverse_obj(fmtl, (0))
+            itag = traverse_obj(fmtl, (1, 0)) or traverse_obj(parse_qs(fmt_url), ('itag', 0))
+            formats.append({
+                'format_id': itag,
+                'url': fmt_url,
+                'ext': 'mp4',
+                'http_headers': {
+                    'referer': 'https://youtube.googleapis.com/',
+                },
+                **self.IF_TABLE.get(int(itag), {}),
+            })
+
+        thumbnail = None
+        title = None
+
+        for block in data:
+            if block is None:
+                continue
+            if isinstance(block, (list, int)):
+                continue
+            if 'ytimg.com' in block:
+                thumbnail = block
+            elif isinstance(block, str):
+                title = block
+            else:
+                continue
 
         return {
-            'id': data.get('iframe_id', token_id),
-            'title': data.get('iframe_id', token_id),
+            'id': title or token_id,
+            'title': title,
+            'thumbnail': thumbnail,
+            'duration': parse_duration(traverse_obj(parse_qs(formats[0]['url']), ('dur', 0))),
             'formats': formats,
-            'thumbnail': data.get('thumbnail'),
-            'duration': parse_duration(traverse_obj(parse_qs(streams[0]['play_url']), ('dur', 0))),
+            **traverse_obj(data, (2, {
+                'title': (-1, {str_or_none}),
+                'thumbnail': (-2, {url_or_none}),
+            })),
         }

From 04725691caf32ff36beb47b250a5abe529554b1e Mon Sep 17 00:00:00 2001
From: 0xvd <0xvd12@gmail.com>
Date: Wed, 25 Feb 2026 00:04:33 +0530
Subject: [PATCH 2/7] Fix wrb regex and bugs

---
 yt_dlp/extractor/blogger.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/yt_dlp/extractor/blogger.py b/yt_dlp/extractor/blogger.py
index 6fc1bd9ee3ba..9f437161ade7 100644
--- a/yt_dlp/extractor/blogger.py
+++ b/yt_dlp/extractor/blogger.py
@@ -82,7 +82,7 @@ def _real_extract(self, url):
             }),
         )
 
-        wrb_data = self._search_regex(r'WcwnYd\s*",\s*"(\[\[.*?\]\])', data, 'wrb.fr')
+        wrb_data = self._search_regex(r'(\[\[.*?"WcwnYd".*?\]\])\s(?:\d+)?', data, 'wrb.fr')
         data = self._parse_json(self._parse_json(wrb_data, None)[0][2], None)
 
         formats = []
@@ -92,7 +92,7 @@ def _real_extract(self, url):
             fmt_url = traverse_obj(fmtl, (0))
             itag = traverse_obj(fmtl, (1, 0)) or traverse_obj(parse_qs(fmt_url), ('itag', 0))
             formats.append({
-                'format_id': itag,
+                'format_id': str(itag),
                 'url': fmt_url,
                 'ext': 'mp4',
                 'http_headers': {

From b055592547298d24c01b3c1f62527cdd14cac738 Mon Sep 17 00:00:00 2001
From: 0xvd <0xvd12@gmail.com>
Date: Wed, 25 Feb 2026 00:07:13 +0530
Subject: [PATCH 3/7] Fix sanitized bug

---
 yt_dlp/extractor/blogger.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/yt_dlp/extractor/blogger.py b/yt_dlp/extractor/blogger.py
index 9f437161ade7..c5bc21324dc5 100644
--- a/yt_dlp/extractor/blogger.py
+++ b/yt_dlp/extractor/blogger.py
@@ -109,9 +109,9 @@ def _real_extract(self, url):
                 continue
             if isinstance(block, (list, int)):
                 continue
-            if 'ytimg.com' in block:
+            if url_or_none(block):
                 thumbnail = block
-            elif isinstance(block, str):
+            elif str_or_none(block):
                 title = block
             else:
                 continue

From 7ccceb61cd1262b62cbd1823f6bd63e36c80991a Mon Sep 17 00:00:00 2001
From: 0xvd <0xvd12@gmail.com>
Date: Wed, 25 Feb 2026 00:10:47 +0530
Subject: [PATCH 4/7] cleanup

---
 yt_dlp/extractor/blogger.py | 11 ++---------
 1 file changed, 2 insertions(+), 9 deletions(-)

diff --git a/yt_dlp/extractor/blogger.py b/yt_dlp/extractor/blogger.py
index c5bc21324dc5..ed31cd9dfabe 100644
--- a/yt_dlp/extractor/blogger.py
+++ b/yt_dlp/extractor/blogger.py
@@ -62,14 +62,10 @@ def _real_extract(self, url):
         token_id = self._match_id(url)
 
         fsid = random.randint(-9999999999999999999, 9999999999999999999)  # Blogger require 19 digit number can be negative or positive
-
         data, _ = self._download_webpage_handle(
             'https://www.blogger.com/_/BloggerVideoPlayerUi/data/batchexecute',
             token_id,
-            note='Downloading blogger batch information',
-            headers={
-                'Referer': 'https://www.blogger.com/',
-            },
+            headers={'Referer': 'https://www.blogger.com/'},
             query={
                 'rpcids': 'WcwnYd',
                 'f.sid': fsid,
@@ -103,11 +99,8 @@ def _real_extract(self, url):
 
         thumbnail = None
         title = None
-
         for block in data:
-            if block is None:
-                continue
-            if isinstance(block, (list, int)):
+            if isinstance(block, (list, int)) or block is None:
                 continue
             if url_or_none(block):
                 thumbnail = block

From ea7180b88fc45cf922de6d21afb966e87aa00e13 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?0x=E2=88=85?= <m.naiemahemad@gmail.com>
Date: Wed, 25 Feb 2026 23:54:33 +0530
Subject: [PATCH 5/7] Apply suggestions from code review

Co-authored-by: dirkf <fieldhouse@gmx.net>
---
 yt_dlp/extractor/blogger.py | 14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)

diff --git a/yt_dlp/extractor/blogger.py b/yt_dlp/extractor/blogger.py
index ed31cd9dfabe..740c2c18faa9 100644
--- a/yt_dlp/extractor/blogger.py
+++ b/yt_dlp/extractor/blogger.py
@@ -62,7 +62,7 @@ def _real_extract(self, url):
         token_id = self._match_id(url)
 
         fsid = random.randint(-9999999999999999999, 9999999999999999999)  # Blogger require 19 digit number can be negative or positive
-        data, _ = self._download_webpage_handle(
+        data = self._download_webpage(
             'https://www.blogger.com/_/BloggerVideoPlayerUi/data/batchexecute',
             token_id,
             headers={'Referer': 'https://www.blogger.com/'},
@@ -78,8 +78,10 @@ def _real_extract(self, url):
             }),
         )
 
-        wrb_data = self._search_regex(r'(\[\[.*?"WcwnYd".*?\]\])\s(?:\d+)?', data, 'wrb.fr')
-        data = self._parse_json(self._parse_json(wrb_data, None)[0][2], None)
+        wrb_data = self._search_json(
+            '', data, 'wrb.fr', token_id, contains_pattern=r'\[\[(?s:.*?)"WcwnYd"(?s:.*?)\]\]',
+            end_pattern=r'\s')
+        data = self._parse_json(wrb_data[0][2], token_id)
 
         formats = []
         for fmtl in traverse_obj(data, (2), default=[]):
@@ -115,8 +117,8 @@ def _real_extract(self, url):
             'thumbnail': thumbnail,
             'duration': parse_duration(traverse_obj(parse_qs(formats[0]['url']), ('dur', 0))),
             'formats': formats,
-            **traverse_obj(data, (2, {
-                'title': (-1, {str_or_none}),
-                'thumbnail': (-2, {url_or_none}),
+            **traverse_obj(data, {
+                'title': (..., {str}, {lambda s: None if url_or_none(s) else s}, any),
+                'thumbnail': (..., {str}, {url_or_none}, any),
             })),
         }

From 61f622aa8b92d81c243229c0b95696a0d79ec61a Mon Sep 17 00:00:00 2001
From: 0xvd <0xvd12@gmail.com>
Date: Sun, 22 Mar 2026 15:15:01 +0530
Subject: [PATCH 6/7] Cleanup, Fix video id, etc.

---
 yt_dlp/extractor/blogger.py | 27 +++++++--------------------
 1 file changed, 7 insertions(+), 20 deletions(-)

diff --git a/yt_dlp/extractor/blogger.py b/yt_dlp/extractor/blogger.py
index 740c2c18faa9..ce073f3937ea 100644
--- a/yt_dlp/extractor/blogger.py
+++ b/yt_dlp/extractor/blogger.py
@@ -52,16 +52,17 @@ class BloggerIE(InfoExtractor):
     # current varianle name (s1) and format table name (r1).
     IF_TABLE = {
         7: {'width': 320, 'height': 240},
+        13: {'width': 256, 'height': 144},  # Source https://gist.github.com/MartinEesmaa/2f4b261cb90a47e9c41ba115a011a4aa#legacy-non-dash
         18: {'width': 640, 'height': 360},
         22: {'width': 1280, 'height': 720},
         37: {'width': 1920, 'height': 1080},
-        13: {'width': 256, 'height': 144},  # Source https://gist.github.com/MartinEesmaa/2f4b261cb90a47e9c41ba115a011a4aa#legacy-non-dash
+        0: {},
     }
 
     def _real_extract(self, url):
         token_id = self._match_id(url)
 
-        fsid = random.randint(-9999999999999999999, 9999999999999999999)  # Blogger require 19 digit number can be negative or positive
+        fsid = random.randint(-9999999999999999999, 9999999999999999999)  # Blogger require 19 digit fsid (number) it can be negative or positive.
         data = self._download_webpage(
             'https://www.blogger.com/_/BloggerVideoPlayerUi/data/batchexecute',
             token_id,
@@ -88,7 +89,7 @@ def _real_extract(self, url):
             if not isinstance(fmtl, list):
                 continue
             fmt_url = traverse_obj(fmtl, (0))
-            itag = traverse_obj(fmtl, (1, 0)) or traverse_obj(parse_qs(fmt_url), ('itag', 0))
+            itag = traverse_obj(fmtl, (1, 0)) or traverse_obj(parse_qs(fmt_url), ('itag', 0), default=0)
             formats.append({
                 'format_id': str(itag),
                 'url': fmt_url,
@@ -99,26 +100,12 @@ def _real_extract(self, url):
                 **self.IF_TABLE.get(int(itag), {}),
             })
 
-        thumbnail = None
-        title = None
-        for block in data:
-            if isinstance(block, (list, int)) or block is None:
-                continue
-            if url_or_none(block):
-                thumbnail = block
-            elif str_or_none(block):
-                title = block
-            else:
-                continue
+        title = traverse_obj(data, (..., {str}, {lambda s: None if url_or_none(s) else s}, any))
 
         return {
-            'id': title or token_id,
+            'id': self._search_regex(r'video-([^-]+)', title, 'video id', default=token_id),
             'title': title,
-            'thumbnail': thumbnail,
             'duration': parse_duration(traverse_obj(parse_qs(formats[0]['url']), ('dur', 0))),
             'formats': formats,
-            **traverse_obj(data, {
-                'title': (..., {str}, {lambda s: None if url_or_none(s) else s}, any),
-                'thumbnail': (..., {str}, {url_or_none}, any),
-            })),
+            'thumbnail': traverse_obj(data, (..., {str}, {url_or_none}, any)),
         }

From ee1dd80566ae8f3598bc09a3cf80608c482ed72b Mon Sep 17 00:00:00 2001
From: 0xvd <0xvd12@gmail.com>
Date: Sun, 22 Mar 2026 15:16:39 +0530
Subject: [PATCH 7/7] cleanup, fix video id extraction, etc.

---
 yt_dlp/extractor/blogger.py | 1 -
 1 file changed, 1 deletion(-)

diff --git a/yt_dlp/extractor/blogger.py b/yt_dlp/extractor/blogger.py
index ce073f3937ea..4149130efa23 100644
--- a/yt_dlp/extractor/blogger.py
+++ b/yt_dlp/extractor/blogger.py
@@ -4,7 +4,6 @@
 from ..utils import (
     parse_duration,
     parse_qs,
-    str_or_none,
     url_or_none,
     urlencode_postdata,
 )
